great_expectations/dataset/autoinspect.py

```diff
@@ -0,0 +1,40 @@
+"""
+Autoinspect utilities to automatically generate expectations by evaluating a dataset.
+"""
+from __future__ import division
+
+import warnings
+from six import string_types
+
+from .util import create_multiple_expectations
+
+
+class AutoInspectError(Exception):
+    """Exception raised for errors in autoinspection.
+
+    Attributes:
+        message -- explanation of the error
+    """
+
+    def __init__(self, message):
+        self.message = message
+
+
+def autoinspect_columns_exist(inspect_dataset):
```

Comment:
Minor point: I don't love this naming convention. It seems likely to devolve into smurf naming.

`ensure_columns_exist`? `check_column_existence`?

Reply:
I don't know smurf naming (but it sounds bad! :))

How about `autoinspect_expect_current_columns_to_exist` (I liked the autoinspect prefix because it's clear that this is not itself an expectation and doesn't itself check anything, but if you were thinking the prefix is no good, I'm open to something else too).

Reply:
Not a strong preference. Here's the ref on smurf naming: https://blog.codinghorror.com/new-programming-jargon/